### DISCUSSION TITLE
Question about unpacking embedded app

### DISCUSSION THREAD

**Discussion by serjum** [php/frankenphp#395] [General]
As I understand from source code and logs, the embedded app is unpacked in `/tmp` directory. Is it possible to avoid unpacking the app?

**Comment by withinboredom** [comment]
Theoretically it could be packed into a phar and embedded in .data. However, php behaves totally different when running in a phar (specifically file access) which will probably be totally unexpected. 

How it works now is actually really similar to how C# apps were packed/unpacked into an executable (and still is) until ahead-of-time compilation was implemented.

I imagine this will be how it works for a long time. In 8.3, we got an actual IR via Opcache, and that could, eventually allow for compiling the code down to the IR, and compiling the VM into the binary and simply executing the IR. We (as in the php community, collectively) are quite a ways from making this possible, but it gets closer with every new version of php.

So, to answer your question. It's likely not possible right now, but can be done, eventually.

**Comment by serjum** [comment]
will be great, thank you!
